pkg/cmd/server/start/command_test.go

```diff
@@ -1,64 +1,46 @@
 package start
 
 import (
+	"errors"
 	"io/ioutil"
 	"os"
+	"path"
 	"strconv"
 	"strings"
 	"testing"
 
 	"github.com/spf13/cobra"
 
-	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
-	utilerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
-
 	configapi "github.com/openshift/origin/pkg/cmd/server/api"
 	configapilatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
-	"github.com/openshift/origin/pkg/cmd/server/api/validation"
 )
 
-func TestCommandBindingListenHttp(t *testing.T) {
-	valueToSet := "http://example.org:9123"
-	masterArgs, masterCfg, masterErr, nodeArgs, nodeCfg, nodeErr := executeAllInOneCommandWithConfigs([]string{"--listen=" + valueToSet})
+// this groups of methods force all the unit tests to share the same config directory
+// the non-cert parts of the directory are cleaned up with every execution, but the certs
+// remain in order to speed up the tests.
+var configDir = ""
 
-	if masterErr != nil {
-		t.Fatalf("Unexpected error: %v", masterErr)
-	}
-	if nodeErr != nil {
-		t.Fatalf("Unexpected error: %v", nodeErr)
+func initConfigDir() {
+	if len(configDir) == 0 {
+		configDir, _ = ioutil.TempDir("", "")
 	}
+}
 
-	if configapi.UseTLS(masterCfg.ServingInfo) {
-		t.Errorf("Unexpected TLS: %v", masterCfg.ServingInfo)
-	}
-	if configapi.UseTLS(masterCfg.AssetConfig.ServingInfo) {
-		t.Errorf("Unexpected TLS: %v", masterCfg.AssetConfig.ServingInfo)
-	}
-	if configapi.UseTLS(nodeCfg.ServingInfo) {
-		t.Errorf("Unexpected TLS: %v", nodeCfg.ServingInfo)
-	}
+func getAllInOneConfigDir() string {
+	return configDir
```

Comment:
actually, can we just rename `initConfigDir` to `getConfigDir` and remove all direct references, to make sure it's initialized as needed?